src/deprecation.cpp

```diff
@@ -12,6 +12,16 @@
 static const std::string CLIENT_VERSION_STR = FormatVersion(CLIENT_VERSION);
 
 void EnforceNodeDeprecation(int nHeight, bool forceLogging) {
+
+    // Do not enforce deprecation in regtest or on testnet
+    const CChainParams& chainparams = Params();
+    std::string networkID = chainparams.NetworkIDString();
+
+    if (networkID != "main") {
```

Comment:
I was a bit dubious about this but, on searching for ``NetworkIDString``, it seems that we do have this style of check in a few other places in ``rpcwallet.cpp``, ``rpcnet.cpp`` and ``init.cpp``.

Reply:
I'll get the test for this done. If there is a more elegant way to reference the current network I'm all ears, this just seemed to be a pretty common way to do it.